src/v/pandaproxy/rest/configuration.cc

```diff
@@ -54,6 +56,50 @@ configuration::configuration()
       "consumer_instance_timeout_ms",
       "How long to wait for an idle consumer before removing it",
       {},
-      std::chrono::minutes{5}) {}
+      std::chrono::minutes{5})
+  , client_cache_max_size(
+      *this,
+      "client_cache_max_size",
+      "The maximum number of kafka clients in the LRU cache",
+      {.needs_restart = config::needs_restart::no},
+      10,
+      [](const size_t max_size) {
+          std::optional<ss::sstring> msg{std::nullopt};
+          if (max_size == 0) {
+              msg = ss::sstring{"Client cache max size must not be zero"};
+          }
+          return msg;
+      })
+  , client_keep_alive(
+      *this,
+      "client_keep_alive",
+      "Time in milliseconds that an idle connection may remain open",
+      {.needs_restart = config::needs_restart::no, .example = "30000"},
+      30s,
+      [](const std::chrono::milliseconds& keep_alive) {
+          std::optional<ss::sstring> msg{std::nullopt};
+          if (keep_alive <= 0s) {
+              msg = ss::sstring{"Client keep alive must be greater than 0"};
+          }
+          return msg;
+      }) {}
+
+config::rest_authn_method configuration::authn_method(size_t listener_idx) {
+    auto pp_api = pandaproxy_api.value();
+
+    if (listener_idx < 0 || listener_idx >= pp_api.size()) {
+        return config::rest_authn_method::none;
+    }
+
+    return pp_api[listener_idx].authn_method.value_or(
+      config::rest_authn_method::none);
+}
+
+// Enables passing a config::binding by value
+// in unit tests.
+configuration& shard_local_cfg() {
+    static thread_local configuration cfg;
+    return cfg;
+}
```

Comment:
I think this is confusing, since it won't be the actual config of a particular instance of `pandaproxy::rest`, it's just a default config. Can the test construct one?